sources/Context.h

```diff
@@ -45,6 +45,13 @@ class Context {
    */
   void process();
 
+ private:
+  /// @brief Slack node origin
+  enum class SlackNodeOrigin {
+    FILE = 0,  ///< Slack node comes from network input file
+    ALGORITHM  ///< Slack node is computed via automatic algorithm
```

Comment:
Maybe we should add a UNKNOWN value to cover potential cases for which we do not assign the origin?

Reply:
We always assign the origin: it depends only of the configuration (with at worst has a default value) and the iidm file, which is required

Reply:
This is true today, what about tomorrow?

Reply:
I don't think the algorithms will change a lot. Besides, for now, we use always the algorithm case since the xml element in the iidm doesn't exist